lib/vnet/ipc_credentials.go

```diff
@@ -0,0 +1,207 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package vnet
+
+import (
+	"crypto"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/utils/keys"
+	"github.com/gravitational/teleport/lib/cryptosuites"
+	"github.com/gravitational/teleport/lib/tlsca"
+)
+
+type ipcCredentials struct {
+	server, client credentials
+}
+
+type credentials struct {
+	trustedCAPEM []byte
+	certPEM      []byte
+	signer       crypto.Signer
+}
+
+func newIPCCredentials() (*ipcCredentials, error) {
+	const (
+		// We don't know which clusters will be connected to at this point so
+		// there's no way to fetch the cluster signature_algorithm_suite or unify
+		// suites across multiple root clusters, so just statically use ECDSA
+		// P-256 for these keys.
+		keyAlgo = cryptosuites.ECDSAP256
```

Comment:
Out of curiosity, would we ever have to customize these? These are only for IPC/local vnet connections, right?

Reply:
You're right, these connections are only on localhost, and the CA keys and the server-side key are in-memory only. I don't really see a reason to customize these, unless some customer security policy demanded it or we want to go post-quantum one day

Reply:
Thanks!